tests/__init__.py

```diff
@@ -160,6 +160,19 @@ def _make_unique(self) -> str:
 }
 
 
-def mksname(purpose: Union[Any], sv: SchemaVersion, f: OutputFormat) -> str:
+_LIMIT_GET_BOM_BY_VERSION_REGEX = re.compile(r'^get_bom_(?P<sv>v(?P<major_version>1)_(?P<minor_version>[0-6]))?(.*)$')
+
+def mksname(purpose: Union[Any], sv: SchemaVersion, f: OutputFormat) -> Optional[str]:
     purpose = purpose if isinstance(purpose, str) else purpose.__name__
+    restrict_to_schema = _LIMIT_GET_BOM_BY_VERSION_REGEX.match(purpose)
+    if restrict_to_schema:
+        mg = restrict_to_schema.groupdict()
+        if mg.get('sv') is not None:
+            # Restrict only to the schema version in the purpose or greater
+            restricted_to_sv = SchemaVersion.from_version(f'{mg.get("major_version")}.{mg.get("minor_version")}')
+            if sv >= restricted_to_sv:
+                return f'{purpose}-{sv.to_version()}.{_SNAME_EXT[f]}'
+            else:
+                return None
```

Comment:
just nope. this functions purpose is to make the name, no matter what.
if you need a delimiter/decider, you should add it elsewhere, for maintainability reasons

Reply:
this extra function would come in handy, when doing the proper filtering at the right position.
see https://github.com/CycloneDX/cyclonedx-python-lib/pull/576#discussion_r1556194919